src/main/java/use_case/EventScheduler.java

```diff
@@ -111,8 +111,9 @@ public boolean checkAvailability(LocalDateTime targetTime, Calendar calendar, Du
                 LocalDateTime endTime = evt.getEndTime().atDate(date);
                 if (targetTime.isAfter(startTime) && targetTime.isBefore(endTime)) {
                     return false;
-                }
-                else if (targetTime.plus(timeNeeded).isAfter(startTime) && targetTime.plus(timeNeeded).isBefore(endTime)) {
+                } else if (targetTime.plus(timeNeeded).isAfter(startTime) && targetTime.plus(timeNeeded).isBefore(endTime)) {
+                    return false;
+                } else if (targetTime.plus(timeNeeded.dividedBy(2)).isAfter(startTime) && targetTime.plus(timeNeeded.dividedBy(2)).isBefore(endTime)) {
```

Comment:
seem to have bug
target time: 7:00 - 9:00
existing event: 8:58 - 8:59
BTW, I think it will be easier to check the cases when return true: target_end =< event_start or target_start >= event_end

Reply:
we need to check for all current events so we can't early return true

added two more else if checks